### PR TITLE
feat: switch over to separate GCP project for snapshot and timestamp

### LINKED ISSUE #349: GA: Organize key permissions for GCP KMS keys
**Description**

1. Remove test & staging keys: contingent on https://github.com/sigstore/root-signing/issues/312
2. Move all online signing keys (the sigstore-root) to two separated GCP projects: one for delegations and one for snapshot and timestamp
3. Create service accounts in each of these projects that are configured with workload identity to access these keys for signing.
4. Create a final service account that only has write access to sync metadata to project-rekor's GCS bucket: https://github.com/sigstore/root-signing/issues/75

Part of https://github.com/sigstore/public-good-instance/issues/376
@var-sdk

### PR DESCRIPTION
**Pull request by asraa**
Signed-off-by: Asra Ali <asraa@google.com>

Fixes https://github.com/sigstore/root-signing/issues/349

* Moves over to a separate GCP project for online keys. 
* Experimental: Create a GH workflow for initializing the repository. This way I won't need to do it on my machine :) 

With this change, log operators, log instances, and on-call maintainers do not have access to roll-back or freeze the TUF repository.

Note: The GCS sync job accounts can stay in the same project: they cannot rollback (timestamps will eventually expire) and they cannot re-sign metadata.

We have this set up with GH repository access so signing can occur through the GH workflows.

### CODE REVIEW

File: .github/workflows/init_repository.yml
Comment (Hayden-IO) [inline .github/workflows/init_repository.yml:47]: To confirm, have these keys been created and does the GitHub Actions workflow have access to them?

File: .github/workflows/init_repository.yml
Comment (Hayden-IO) [inline .github/workflows/init_repository.yml]: can you add me?

Comment (Hayden-IO) [reply .github/workflows/init_repository.yml]: and any other folks who are active with write access? same with snapshot-timestamp.yml

File: .github/workflows/snapshot-timestamp.yml
Comment (Hayden-IO) [inline .github/workflows/snapshot-timestamp.yml:53]: can we make this change yet? don't we need a new root signing event since we're switching to a new key?

File: .github/workflows/init_repository.yml
Comment (asraa) [reply .github/workflows/init_repository.yml:47]: Yep

File: .github/workflows/snapshot-timestamp.yml
Comment (asraa) [reply .github/workflows/snapshot-timestamp.yml:53]: the new root signing event is happening today! so i need these triggers

File: .github/workflows/init_repository.yml
Comment (asraa) [reply .github/workflows/init_repository.yml]: sure: not totally concerned since it opens a PR and anyone can/should review

**Review by Hayden-IO — APPROVED** [review]

**Comment by asraa** [comment]
cc @var-sdk

**Pull request merged**
